tests_e2e/tests/ext_sequencing/ext_sequencing.py

```diff
@@ -0,0 +1,211 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# This test adds extensions with multiple dependencies to a VMSS using the 'provisionAfterExtensions' property and
+# validates they are enabled in order of dependencies.
+#
+import json
+import random
+from datetime import datetime
+from pathlib import Path
+from typing import List, Dict, Any
+
+from assertpy import fail
+from azure.core.exceptions import HttpResponseError
+from azure.mgmt.compute.models import VirtualMachineExtensionInstanceView
+
+from tests_e2e.tests.lib.agent_test import AgentTest, TestSkipped
+from tests_e2e.tests.lib.agent_test_context import AgentTestContext
+from tests_e2e.tests.lib.identifiers import VmExtensionIds, VmssIdentifier, VmIdentifier
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.ssh_client import SshClient
+from tests_e2e.tests.lib.virtual_machine_client import VirtualMachineClient
+from tests_e2e.tests.lib.virtual_machine_scale_set_client import VirtualMachineScaleSetClient
+
+
+class ExtSequencing(AgentTest):
+    def __init__(self, context: AgentTestContext):
+        super().__init__(context)
+        self._ssh_client = self._context.create_ssh_client()
+
+    def get_sorted_extension_names(self, extensions: List[VirtualMachineExtensionInstanceView], ssh_client: SshClient):
+        # Log the extension enabled datetime
+        for ext in extensions:
+            enabled_time = ssh_client.run_command(f"ext_sequencing-get_ext_enable_time.py --ext_type {ext.type}", use_sudo=True)
+            ext.time = datetime.strptime(enabled_time.replace('\n', ''), u'%Y-%m-%d %H:%M:%S')
+            log.info("Extension {0} enabled time: {1}".format(ext.name, ext.time))
+
+        # sort the extensions based on their enabled datetime
+        sorted_extensions = sorted(extensions, key=lambda ext_: ext_.time)
+        log.info("Sorted extension names with time: {0}".format(
+            ', '.join(["{0}: {1}".format(ext.name, ext.time) for ext in sorted_extensions])))
+        sorted_extension_names = [ext.name for ext in sorted_extensions]
+        return sorted_extension_names
+
+    @staticmethod
+    def delete_extension_with_dependencies_should_fail(vmss: VirtualMachineScaleSetClient, dependency_map: dict):
+        # Randomly choose an extension that has dependencies on it to remove
+        ext_to_remove = None
+        for ext in dependency_map:
+            if dependency_map[ext]:
+                ext_to_remove = random.choice(dependency_map[ext])
+                break
+        if ext_to_remove:
+            try:
+                vmss.delete_extension(ext_to_remove)
+            except HttpResponseError as e:
+                if "(BadRequest)" in e.message and "provisionAfterExtensions" in e.message:
```

Comment:
When you try to remove an extension with dependencies CRP reports an error which we catch here
```
{
  "innererror": {
    "internalErrorCode": "MissingDependentExtensionInArgumentDependentExtensionExistsInVMSSInstance"
  },
  "code": "BadRequest",
  "message": "On resource 'lisawalinuxagentextseq20230923164508e0', extension 'RunCommand' specifies 'VmAccess' in its provisionAfterExtensions property, but the extension 'VmAccess' will no longer exist. First, remove the extension 'RunCommand' or remove 'VmAccess' from the provisionAfterExtensions property of 'RunCommand'."
}
```